src/driver/mongodb.rs

```diff
@@ -22,7 +22,7 @@ impl Driver for MongoDriver {
   fn connect(uri: &str) -> Result<Self, Error> {
     let config = try!(connstring::parse(uri));
 
-    if let Some(db_name) = config.clone().database {
+    if let Some(db_name) = config.database.clone() {
```

Comment:
Tried two variants:

```rust
if let Some(db_name) = config.database.as_ref() {
  Ok(MongoDriver {
    database: try!(Client::with_config(config, None, None)).db(db_name)
  })
}
```

and 

```rust
if let Some(ref db_name) = config.database {
  Ok(MongoDriver {
    database: try!(Client::with_config(config, None, None)).db(db_name)
  })
}
```

Both don't work. This bit of code is inconsequential enough, and I moved the `clone` for performance optimizations, so I really don't care enough to remove `clone` here and I'm not even sure it can be done.